extensions/amp-story/1.0/amp-story.js

```diff
@@ -1215,6 +1218,10 @@ export class AmpStory extends AMP.BaseElement {
     } else if (direction === NavigationDirection.NEXT) {
       this.storyNavigationPath_.push(targetPage);
     }
+    setHistoryState(
+        this.win,
+        HistoryState.NAVIGATION_PATH,
+        this.storyNavigationPath_.map(page => page.element.id));
```

Comment:
Sorry to be picky on this, but we try to be extra careful with all the code that runs during navigation, since it delays the time where the next page will be visible.
Should we fill the `this.storyNavigationPath_` array with page IDs in the first place, and do the `getPageByById()` in the `switchTo` method, using the ID this method returns?
This way we could skip that loop and just store `this.storyNavigationPath_`